/*
 * Engineering Ingegneria Informatica S.p.A.
 *
 * Copyright (C) 2023 Regione Emilia-Romagna
 * <p/>
 * This program is free software: you can redistribute it and/or modify it under the terms of
 * the GNU Affero General Public License as published by the Free Software Foundation,
 * either version 3 of the License, or (at your option) any later version.
 * <p/>
 * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
 * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
 * See the GNU Affero General Public License for more details.
 * <p/>
 * You should have received a copy of the GNU Affero General Public License along with this program.
 * If not, see <https://www.gnu.org/licenses/>.
 */

package it.eng.hsm.prod.itagile.firmaremota.ws;

import javax.xml.ws.WebFault;

/**
 * This class was generated by Apache CXF 2.7.14.redhat-1 2017-08-29T19:04:40.516+02:00 Generated source version:
 * 2.7.14.redhat-1
 */
@WebFault(name = "fault", targetNamespace = "http://ws.firmaremota.itagile.it")
public class RemoteSignatureExceptionException extends Exception {

    private it.eng.hsm.prod.itagile.firmaremota.ws.RemoteSignatureException fault;

    public RemoteSignatureExceptionException() {
        super();
    }

    public RemoteSignatureExceptionException(String message) {
        super(message);
    }

    public RemoteSignatureExceptionException(String message, Throwable cause) {
        super(message, cause);
    }

    public RemoteSignatureExceptionException(String message,
            it.eng.hsm.prod.itagile.firmaremota.ws.RemoteSignatureException fault) {
        super(message);
        this.fault = fault;
    }

    public RemoteSignatureExceptionException(String message,
            it.eng.hsm.prod.itagile.firmaremota.ws.RemoteSignatureException fault, Throwable cause) {
        super(message, cause);
        this.fault = fault;
    }

    public it.eng.hsm.prod.itagile.firmaremota.ws.RemoteSignatureException getFaultInfo() {
        return this.fault;
    }
}
